### PR TITLE
[Gluten-986] Remove unimportant metrics in Agg and Join operator and input metrics for all operator.

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Agg should not care about preProject, postProject and extractionNeeded's metrics, these nodes does not modify row numbers, maybe change total bytes. We should keep it clearly and remove these.

Operator's input metrics and output metrics are equals, we just need output metrics and align with Spark.

https://github.com/facebookincubator/velox/blob/db4ea520893c30ac9197e75b88a3b85bfc801758/velox/exec/Driver.cpp#L346-L373

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by FelixYBW** [comment]
Some of the metrics are used in our analysis tool. Let me check later.

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 1]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
-    "buildOutputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of build side output rows"),
-    "buildOutputVectors" -> SQLMetrics.createMetric(
-      sparkContext, "number of build side output vectors"),
-    "buildOutputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of build side output bytes"),
-    "buildCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "build side cpu wall time count"),
-    "buildWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_build_input"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:187]: We need this. It actually count the arrow2Velox conversion

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 2]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
-    "streamOutputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of stream side output rows"),
-    "streamOutputVectors" -> SQLMetrics.createMetric(
-      sparkContext, "number of stream side output vectors"),
-    "streamOutputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of stream side output bytes"),
-    "streamCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "stream side cpu wall time count"),
-    "streamWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_stream_input"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:135]: keep this, change to totaltime stream input

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 3]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
-    "streamVeloxToArrow" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_velox_to_arrow_converter"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:137]: keep it, it only count part of the conversion

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 4]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
-    "streamPreProjectionOutputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of stream preProjection output rows"),
-    "streamPreProjectionOutputVectors" -> SQLMetrics.createMetric(
-      sparkContext, "number of stream preProjection output vectors"),
-    "streamPreProjectionOutputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of stream preProjection output bytes"),
-    "streamPreProjectionCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "stream preProjection cpu wall time count"),
-    "streamPreProjectionWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_stream_preProjection"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:162]: keep this

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 5]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
-    "buildPreProjectionOutputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of build preProjection output rows"),
-    "buildPreProjectionOutputVectors" -> SQLMetrics.createMetric(
-      sparkContext, "number of build preProjection output vectors"),
-    "buildPreProjectionOutputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of build preProjection output bytes"),
-    "buildPreProjectionCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "build preProjection cpu wall time count"),
-    "buildPreProjectionWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_build_preProjection"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:212]: keep it

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 6]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
       sparkContext, "number of hash build output vectors"),
     "hashBuildOutputBytes" -> SQLMetrics.createSizeMetric(
       sparkContext, "number of hash build output bytes"),
-    "hashBuildCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "hash build cpu wall time count"),
+    "hashBuildCpuNanos" -> SQLMetrics.createNanoTimingMetric(
+      sparkContext, "hash build cpu time"),
     "hashBuildWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_hashbuild"),
+      sparkContext, "hash build wall time"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:125]: change to totaltime of hash build

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 7]
```diff
@@ -248,26 +136,16 @@ trait HashJoinLikeExecTransformer
...
       sparkContext, "number of hash probe output vectors"),
     "hashProbeOutputBytes" -> SQLMetrics.createSizeMetric(
       sparkContext, "number of hash probe output bytes"),
-    "hashProbeCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "hash probe cpu wall time count"),
+    "hashProbeCpuNanos" -> SQLMetrics.createNanoTimingMetric(
+      sparkContext, "hash probe cpu time"),
     "hashProbeWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_hashprobe"),
+      sparkContext, "hash probe wall time"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:148]: totaltime of probe

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 8]
```diff
@@ -285,103 +163,16 @@ trait HashJoinLikeExecTransformer
...
-    "postProjectionOutputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of postProjection output rows"),
-    "postProjectionOutputVectors" -> SQLMetrics.createMetric(
-      sparkContext, "number of postProjection output vectors"),
-    "postProjectionOutputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of postProjection output bytes"),
-    "postProjectionCount" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "postProjection cpu wall time count"),
-    "postProjectionWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_postProjection"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:307]: keep it, use totaltime of postProjection

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 9]
```diff
@@ -113,128 +113,16 @@ trait HashJoinLikeExecTransformer
...
-      sparkContext, "build preProjection cpu wall time count"),
-    "buildPreProjectionWallNanos" -> SQLMetrics.createNanoTimingMetric(
-      sparkContext, "totaltime_build_preProjection"),
-    "buildPreProjectionPeakMemoryBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "build preProjection peak memory bytes"),
-    "buildPreProjectionNumMemoryAllocations" -> SQLMetrics.createMetric(
-      sparkContext, "number of build preProjection memory allocations"),
-
-    "hashBuildInputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of hash build input rows"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:219]: keep this. Then we needn't to check the previous operator to get the input rows

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 10]
```diff
@@ -248,26 +136,16 @@ trait HashJoinLikeExecTransformer
     "hashBuildSpilledFiles" -> SQLMetrics.createMetric(
       sparkContext, "total spilled files of hash build"),
 
-    "hashProbeInputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of hash probe input rows"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:252]: the same, let keep this one also

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 11]
```diff
@@ -248,26 +136,16 @@ trait HashJoinLikeExecTransformer
...
-    "hashProbeRawInputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of hash probe raw input rows"),
-    "hashProbeRawInputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of hash probe raw input bytes"),
     "hashProbeOutputRows" -> SQLMetrics.createMetric(
       sparkContext, "number of hash probe output rows"),
     "hashProbeOutputVectors" -> SQLMetrics.createMetric(
       sparkContext, "number of hash probe output vectors"),
     "hashProbeOutputBytes" -> SQLMetrics.createSizeMetric(
       sparkContext, "number of hash probe output bytes"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:144]: looks it's broken. let's remove it.
<html>
<body>


number of hash probe output bytes | -39,682,000,181,513,076,736 (495,594,061,042,182,848.0)
-- | --



</body>
</html>

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 12]
```diff
@@ -285,103 +163,16 @@ trait HashJoinLikeExecTransformer
...
-    "postProjectionInputVectors" -> SQLMetrics.createMetric(
-      sparkContext, "number of postProjection input vectors"),
-    "postProjectionInputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of postProjection input bytes"),
-    "postProjectionRawInputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of postProjection raw input rows"),
-    "postProjectionRawInputBytes" -> SQLMetrics.createSizeMetric(
-      sparkContext, "number of postProjection raw input bytes"),
-    "postProjectionOutputRows" -> SQLMetrics.createMetric(
-      sparkContext, "number of postProjection output rows"),
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:299]: let's keep it. It's the final output rows from hash join

File: gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala [hunk 13]
```diff
@@ -285,103 +163,16 @@ trait HashJoinLikeExecTransformer
...
-    "postProjectionOutputVectors" -> SQLMetrics.createMetric(
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala:300]: and this one

**Comment by FelixYBW** [comment]
we can remove the other metrics

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
@waitinfuture OK to me.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**